### PR TITLE
Change signature of Node#getPublicKey from byte[] to PublicKey [ECR-2758]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Change signature of Node#getPublicKey from byte[] to PublicKey.

---
See: https://jira.bf.local/browse/ECR-2758

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding-core/src/main/java/com/exonum/binding/service/NodeFake.java [hunk 1]
```diff
@@ -57,15 +58,15 @@
...
+  private final PublicKey publicKey;
 
   /**
    * Creates a new node fake with the given database and an empty public key.
    *
    * @param database a database to provide snapshots of
    */
   public NodeFake(MemoryDb database) {
-    this(database, new byte[0]);
+    this(database, PublicKey.fromBytes(new byte[0]));
```
Comment (bullet-tooth) [inline exonum-java-binding-core/src/main/java/com/exonum/binding/service/NodeFake.java]: `PublicKey.fromBytes(new byte[PUBLIC_KEY_BYTES])`

File: exonum-java-binding-core/src/main/java/com/exonum/binding/service/NodeProxy.java [hunk 2]
```diff
@@ -116,8 +117,9 @@ private static native void nativeSubmit(long nodeHandle, UserTransactionAdapter
    * @throws IllegalStateException if the node proxy is closed
    */
   @Override
-  public byte[] getPublicKey() {
-    return nativeGetPublicKey(getNativeHandle());
+  public PublicKey getPublicKey() {
+    byte[] bytePk = nativeGetPublicKey(getNativeHandle());
```
Comment (bullet-tooth) [inline exonum-java-binding-core/src/main/java/com/exonum/binding/service/NodeProxy.java]: I wouldn't use `byte` prefix in the name. Consider: `key`, `publicKey`

File: exonum-java-binding-core/src/main/java/com/exonum/binding/service/Node.java [hunk 3]
```diff
@@ -56,5 +56,5 @@ void submitTransaction(Transaction transaction)
    *
    * @throws IllegalStateException if the node proxy is closed
    */
-  byte[] getPublicKey();
+  PublicKey getPublicKey();
```
Comment (dmitry-timofeev) [inline exonum-java-binding-core/src/main/java/com/exonum/binding/service/Node.java]: A part of this task is to improve documentation

> Also clarify which of the two public keys it returns and its relationship to Blockchain#getActiveConfiguration.

Comment (MakarovS) [reply exonum-java-binding-core/src/main/java/com/exonum/binding/service/Node.java]: You mean service vs consensus public key? I've added this to the Javadoc.
As I understand these keys are stored in `StoredConfiguration#validatorKeys`, should I mention that in `Node#getPublicKey` Javadoc?

**Pull request merged**